peernode/src/peer-node/peer/mod.rs

```diff
@@ -0,0 +1,135 @@
+use orcanet_market::{bridge::spawn, Config, Peer};
+
+use proto::market::{FileInfo, HoldersResponse, User};
+
+use anyhow::{anyhow, Result};
+
+use std::collections::HashMap;
+
+// test market
+#[cfg(feature = "test_local_market")]
+pub struct MarketClient {
+    local: HashMap<String, HoldersResponse>,
+    //// update in later commit
+    //local: HashMap<FileInfoHash, HoldersResponse>,
+}
+#[cfg(feature = "test_local_market")]
+impl MarketClient {
+    pub async fn new(_config: Config) -> Result<Self> {
+        Ok(MarketClient {
+            local: Default::default(),
+        })
+    }
+    pub async fn check_holders(&mut self, file_info_hash: String) -> Result<HoldersResponse> {
+        //// update in later commit
+        //pub async fn check_holders(&mut self, file_info_hash: FileInfoHash) -> Result<HoldersResponse> {
+        match self.local.get(&file_info_hash) {
+            Some(res) => Ok(res.clone()),
+            None => Err(anyhow!("not found!")),
+        }
+    }
+
+    pub async fn register_file(
+        &mut self,
+        id: String,
+        name: String,
+        ip: String,
+        port: i32,
+        price: i64,
+        file_hash: String,
+    ) -> Result<()> {
+        //// update in later commit
+        //pub async fn register_file(
+        //    &mut self,
+        //    user: User,
+        //    file_info_hash: FileInfoHash,
+        //    file_info: FileInfo,
+        //) -> Result<()> {
+        let user = User {
+            id,
+            name,
+            ip,
+            port,
+            price,
+        };
+        match self.local.get_mut(&file_hash) {
+            Some(res) => {
+                res.holders.retain(|u| u.id != user.id);
+                res.holders.push(user.clone());
+                Ok(())
+            }
+            None => {
+                self.local.insert(
+                    file_hash.clone(),
+                    HoldersResponse {
+                        file_info: Some(FileInfo {
+                            file_hash,
+                            chunk_hashes: vec![],
+                            file_size: 0,
+                            file_name: "foo".into(),
```

Comment:
There is a file_hash -> file_name map in `config.get_file_names()`

Reply:
The test local market will be updated to use the new interface in a later PR, so I didn't include that in this PR.